### PR TITLE
chore: update go-libp2p to v0.26.1

### PR DESCRIPTION
**Pull request by marten-seemann**
[v0.26.1](https://github.com/libp2p/go-libp2p/releases/tag/v0.26.1) contains two fixes that should probably make it into the final release.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -172,7 +172,7 @@ require (
 	github.com/libp2p/go-buffer-pool v0.1.0 // indirect
 	github.com/libp2p/go-cidranger v1.1.0 // indirect
 	github.com/libp2p/go-flow-metrics v0.1.0 // indirect
-	github.com/libp2p/go-libp2p-asn-util v0.2.0 // indirect
+	github.com/libp2p/go-libp2p-asn-util v0.3.0 // indirect
```
Comment (marten-seemann) [inline go.mod:175]: Not sure why the Go toolchain decided to update this dependency. This update is not part of the v0.26.1 release.

**Review by Jorropo — APPROVED** [review]
LGTM Ty

**Comment by marten-seemann** [comment]
Note: we'll have to revert this change. Unified CI's release workflow is broken, and it cut the release on the wrong branch: https://github.com/protocol/.github/issues/481

**Comment by Jorropo** [comment]
@marten-seemann ok just open a new PR with a revert and the correct update.
Btw do:
```go
find -type f -name "go.mod" | while read p; do cd $(dirname $p); go mod tidy; cd -; done
```
to correctly fix the `go.mod` files in kubo :slightly_smiling_face: .

**Comment by marten-seemann** [comment]
> Btw do:
> 
> ```shell
> find -type f -name "go.mod" | while read p; do cd $(dirname $p); go mod tidy; cd -; done
> ```
> 
> to correctly fix the `go.mod` files in kubo 🙂 .

Isn't it beautiful? ;)

**Pull request merged**